### PR TITLE
[PDI-19937]-Database Password Field shows password in plain text when searching job's metadata

### PR DESCRIPTION
**Pull request by SahithiKommagani6**
[PDI-19937]-Database Password Field shows password in plain text when searching job's metadata

[PDI-19937]: https://hv-eng.atlassian.net/browse/PDI-19937?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by buildguy** [comment]
### :white_check_mark: Build finished in 59m 24s
Build command:
```console
mvn clean verify -B -e -Daudit -amd -pl engine
```
---
:ok_hand: All tests passed!
 
Tests run: **4092**, Failures: **0**, Skipped: **13** &nbsp;&nbsp; [Test Results](https://wingman.orl.eng.hitachivantara.com/job/projects/job/pdi/job/PR-9063/1/testReport)

 ---
 
 

:information_source:  _This is an automatic message_

File: engine/src/main/java/org/pentaho/di/job/JobMeta.java [hunk 1]
```diff
@@ -2181,7 +2182,7 @@ public List<StringSearchResult> getStringList( boolean searchSteps, boolean sear
         // if ( includePasswords )
         // {
         if ( meta.getPassword() != null ) {
-          stringList.add( new StringSearchResult( meta.getPassword(), meta, this,
+          stringList.add( new StringSearchResult( Strings.repeat("*", meta.getPassword().length()), meta, this,
```
Comment (peterrinehart) [inline engine/src/main/java/org/pentaho/di/job/JobMeta.java:2185]: Change is fine, please update the copyright header in this file.

**Review by peterrinehart — CHANGES_REQUESTED** [review]

File: engine/src/main/java/org/pentaho/di/job/JobMeta.java (hunk 1 above)
Comment (SahithiKommagani6) [reply engine/src/main/java/org/pentaho/di/job/JobMeta.java:2185]: updated copyright header

**Comment by buildguy** [comment]
### :white_check_mark: Build finished in 1h 3m 45s
Build command:
```console
mvn clean verify -B -e -Daudit -amd -pl engine
```
---
:ok_hand: All tests passed!
 
Tests run: **4092**, Failures: **0**, Skipped: **13** &nbsp;&nbsp; [Test Results](https://wingman.orl.eng.hitachivantara.com/job/projects/job/pdi/job/PR-9063/2/testReport)

 ---
 
 

:information_source:  _This is an automatic message_

**Pull request merged**